src/message.ts

```diff
@@ -119,32 +119,39 @@ export async function verifyMessageSignature(message: Message, didResolver: DIDR
 
   // use a set to do O(1) "lookups" for the correct id
   const verificationMethodIds = new Set();
-  let publicKey;
+  let verificationMethod: VerificationMethod;
 
-  for (let verificationMethod of verificationMethods) {
-    verificationMethodIds.add(verificationMethod.id);
+  for (const vm of verificationMethods) {
+    verificationMethodIds.add(vm.id);
```

Comment:
Sanity: is the set of ID necessary? Can't we just compare against `vm.id` directly?

Reply:
@thehenrytsai definitely not necessary. It's potentially an overzealous optimization. Using a set to compare for equality makes each comparison `O(1)` vs. `O(n)`. Since we're looping over an unbounded array, I thought it might be worth leveraging O(1) lookups. As i was looking at this again, we could even remove the use of `add` by refactoring the optimization like so:

```js
// use a set to do O(1) lookups for the correct id
const kidSet = new Set();
let verificationMethod: VerificationMethod;

for (const vm of verificationMethods) {
  if (kidSet.has(vm.id)) {
    verificationMethod = vm;
    break;
  }
}
```

Happy to switch to direct comparison if you think the optimization is unnecessary. Lmk what you think